webapp/src/main/webapp/WEB-INF/conf/officeonline/configuration.xml

```diff
@@ -24,13 +24,30 @@
   xmlns="http://www.exoplatform.org/xml/ns/kernel_1_2.xsd">
 
   <component>
-    <key>org.exoplatform.officeonline.OfficeOnlineEditorService</key>
-    <type>org.exoplatform.officeonline.OfficeOnlineEditorService</type>
+    <key>org.exoplatform.officeonline.rest.WOPIResource</key>
+    <type>org.exoplatform.officeonline.rest.WOPIResource</type>
   </component>
+
+  <component>
+    <key>org.exoplatform.officeonline.rest.EditorResource</key>
+    <type>org.exoplatform.officeonline.rest.EditorResource</type>
+  </component>
+
+
+  <component>
+    <key>org.exoplatform.officeonline.WOPIService</key>
+    <type>org.exoplatform.officeonline.WOPIService</type>
+  </component>
+  <component>
+    <key>org.exoplatform.officeonline.EditorService</key>
+    <type>org.exoplatform.officeonline.EditorService</type>
+  </component>
+
+
   
   <!-- WOPIDiscoveryPlugin -->
   <external-component-plugins>
-    <target-component>org.exoplatform.officeonline.OfficeOnlineEditorService</target-component>
+    <target-component>org.exoplatform.officeonline.WOPIService</target-component>

```

Comment:
WOPIService misleading being a plugin of another service. WOPIPlugin or WOPIConnector or WOPIProvider will be better name.

Reply:
WOPIService is not a part of another service.
We've confirmed architecture, where we have AbstractOfficeOnlineService which contains all common methods for working with users/files, and two container services - WOPIService and EditorService with specific logic.

Reply:
If so, it was about outdated code and not a case anymore.